### ISSUE TITLE
Bug: Uneven vertical alignment when copy pasting from other source and subsequent cases of random deletion of characters

### ISSUE THREAD

**Issue by Khris777** [spyder-ide/spyder#7910]
## Problem Description

If I take code from another source like Notepad++ using one tab per identation level and paste it into Spyder, the characters aren't aligned vertically.

The following image shows first some code that has been entered directly into Spyder and below the same code copypasted from Notepad++. You can see how the letters aren't properly aligned in the vertical anymore.

![image](https://user-images.githubusercontent.com/10170372/45625535-16ac9700-ba5b-11e8-9a42-c7f0e22ca944.png)

Furthermore odd things start happening when editing that non-aligned code, e.g. when following these steps:

1. Outside of Spyder in Notepad++ indent the print-statement with another tab, so that it is two levels deeper than the prior line.
2. Paste into Spyder.
3. In Spyder using backspace remove two levels of indentation bringing the print-statement to the same level as the prior for-loop.
4. Press tab once. This for whatever reason deletes the first three letters of the print-statement.

See this image for what steps 2-4 look like for me.

![image](https://user-images.githubusercontent.com/10170372/45625644-5e332300-ba5b-11e8-879d-24464a47a9a1.png)

Another example:

If you go back to step 2, then indent the print-statement in Spyder yet another level (cursor being directly before the p) and press backspace the whole statement gets deleted.

This not just happens with code from Notepad++ but also when trying other sources like Windows Code Writer or MS Word.

Example Code from Sypder:

```
for i in [1,2,3]:
    for j in [4,5,6]:
        for k in [7,8,9]:
            print(i+j+k)
```

The same code directly copypasted from Notepad++:
```
for i in [1,2,3]:
	for j in [4,5,6]:
		for k in [7,8,9]:
			print(i+j+k)
```

It seems to be a problem of mixing and/or converting spaces and tabs.

## Versions
* Spyder version: 3.3.0
* Python version: 3.6.6
* Qt version: 5.9.4
* PyQt version: 5.9.2
* Operating System name/version: Windows 10

### Dependencies
```
IPython >=4.0     :  6.5.0 (OK)
cython >=0.21     :  0.28.4 (OK)
jedi >=0.9.0      :  0.12.0 (OK)
matplotlib >=2.0.0:  2.2.3 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
numpy >=1.7       :  1.14.5 (OK)
pandas >=0.13.1   :  0.23.4 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pyflakes >=0.6.0  :  1.6.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  1.8.4 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
rope >=0.9.4      :  0.10.7 (OK)
sphinx >=0.6.6    :  1.7.4 (OK)
sympy >=0.7.3     :  1.2 (OK)
```

**Comment by jnsebgosselin** [comment]
Ok, I cannot reproduce it in 3.x or Master.

@ccordoba12 I think this is a duplicate of #7432 and was fixed in Spyder v3.3.1.

@Khris777 Could you update Spyder and check if the issue is resolved please?

**Comment by ccordoba12** [comment]
Thanks Jean. Closing as a duplicate of #7678 then.
